include/ignition/math/AdditivelySeparableScalarField3.hh

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef IGNITION_MATH_SEPARABLE_SCALAR_FIELD3_HH_
+#define IGNITION_MATH_SEPARABLE_SCALAR_FIELD3_HH_
+
+#include <limits>
+#include <utility>
+
+#include <ignition/math/Region3.hh>
+#include <ignition/math/Vector3.hh>
+#include <ignition/math/config.hh>
+
+namespace ignition
+{
+  namespace math
+  {
+    // Inline bracket to help doxygen filtering.
+    inline namespace IGNITION_MATH_VERSION_NAMESPACE {
+    //
+    /** \class AdditivelySeparableScalarField3\
+     * AdditivelySeparableScalarField3.hh\
+     * ignition/math/AdditivelySeparableScalarField3.hh
+     */
+    /// \brief The AdditivelySeparableScalarField3 class constructs
+    /// a scalar field F in R^3 as a sum of scalar functions i.e.
+    /// F(x, y, z) = k (p(x) + q(y) + r(z)).
+    ///
+    /// ## Example
+    ///
+    /// \snippet examples/additively_separable_scalar_field3_example.cc complete
+    template<typename ScalarFunctionT, typename ScalarT>
+    class AdditivelySeparableScalarField3
+    {
+      /// \brief Constructor
+      /// \param[in] _k scalar constant
+      /// \param[in] _p scalar function of x
+      /// \param[in] _q scalar function of y
+      /// \param[in] _r scalar function of z
+      public: AdditivelySeparableScalarField3(
+          ScalarT _k, ScalarFunctionT _p,
+          ScalarFunctionT _q, ScalarFunctionT _r)
+      : k(_k), p(std::move(_p)), q(std::move(_q)), r(std::move(_r))
+      {
+      }
+
+      /// \brief Evaluate the scalar field at `_point`
+      /// \param[in] _point scalar field argument
+      /// \return the result of evaluating `F(_point)`
+      public: ScalarT Evaluate(const Vector3<ScalarT> &_point) const
+      {
+        return this->k * (
+            this->p(_point.X()) +
+            this->q(_point.Y()) +
+            this->r(_point.Z()));
+      }
+
+      /// \brief Call operator overload
+      /// \see SeparableScalarField3::Evaluate()
+      public: ScalarT operator()(const Vector3<ScalarT> &_point) const
```

Comment:
Interestingly doxygen didn't complain about the missing `param[in]` and `return`

Reply:
Would you like to add them? It's a transparent forwarding overload, for convenience only.

Reply:
Yeah I think it's worth it for completeness, thanks!

Reply:
Would you add them? It'll be an exact copy of `Evaluate()` documentation.